cmd/wait/runner.go

```diff
@@ -285,5 +306,49 @@ func (r *runner) run(ctx context.Context, _ *cobra.Command, _ []string) error {
 		r.logger.LogCtx(ctx, "message", "external-dns is ready")
 	}
 
+	// Wait for all charts to be deployed
+	r.logger.LogCtx(ctx, "message", "waiting for all chart CRs to be in deployed state")
+
+	o := func() error {
+		charts := applicationv1alpha1.ChartList{}
+		err = ctrlClient.List(ctx, &charts, client.InNamespace("giantswarm"))
+		if err != nil {
+			r.logger.LogCtx(ctx, "message", fmt.Sprintf("cannot list charts in namespace giantswarm: %s", err))
+			return microerror.Mask(notReadyError)
+		}
+
+		if len(charts.Items) < 2 {
+			r.logger.LogCtx(ctx, "message", fmt.Sprintf("Waiting for at least 2 Chart CRs to exist in giantswarm namespace, found %d", len(charts.Items)))
+			return microerror.Mask(notReadyError)
+		}
+
+		deployedCount := 0
+		notDeployed := make([]string, 0)
+		for _, chart := range charts.Items {
+			if chart.Status.Release.Status == "deployed" {
+				deployedCount = deployedCount + 1
+			} else {
+				notDeployed = append(notDeployed, chart.Name)
+			}
```

Comment:
If you care about the number of charts, then ignore this comment.

Otherwise, you could just return an error if `chart.Status.Release.Status != "deployed"`?

Reply:
to have some feedback over the progress I'd like to have the number of pending charts printed